### PR TITLE
Further Time Tacking Improvement

### LINKED ISSUE #4076: Improve time tracking
See https://discuss.webknossos.org/t/a-few-updates-for-the-time-tracking-tool/1374

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR adds the following suggest improvements to the time tracking view:
- [x] The format of the duration in the tooltip is now minutes:seconds
- [x] A spinning wheel is showing up when the data is being fetched from the backend and rendered into a diagram
- [x] There seems to be a bug that causes tooltips to not being rendered. Look at the [discuss](https://discuss.webknossos.org/t/a-few-updates-for-the-time-tracking-tool/1374/8?u=potto) for further clarification  

### URL of deployed dev instance (used for testing):
- [https://improvetimetracking.webknossos.xyz](https://improvetimetracking.webknossos.xyz)

### Steps to test:
- Get some tracking data by e.g. tracing.
- open the time tracking view and select the correct user and date.
- When the data is fetched by the server, there should be a spinning wheel visible. But it is not visible, while the google charts library is actually rendering the chart.
- The tooltip should always be next to the mouse (if visible). It should **not** appear somewhere else and then "beam" next to the mouse.
- The duration displayed in the tooltip should have the format of `minutes:seconds`.
- Also please consider that the tooltip might be partly positioned out of the window. I added a fix that adjusts the position for these cases.  Please also try to whether you can find a case where the tooltip is still partly placed outside the window.

### Issues:
- fixes #4076

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

**Comment by MichaelBuessemeyer** [comment]
@philippotto This PR is ready for your first review :slightly_smiling_face:

File: frontend/javascripts/admin/time/time_line_chart_view.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  componentWillUnmount() {
+    if (this.chartScrollElement) {
+      this.chartScrollElement.removeEventListener("mousemove", this.adjustTooltipPosition);
+    }
+  }
+
+  // We need to adjust the tooltips position manually because it is not positioned correctly when scrolling down.
+  // This fix was suggested by
+  // https://stackoverflow.com/questions/52755733/google-charts-tooltips-have-wrong-position-when-inside-a-scrolling-container.
+  applyTooltipPositioningFix = () => {
```
Comment (philippotto) [inline frontend/javascripts/admin/time/time_line_chart_view.js]: Did you adapt the fix now? Last time we talked, there was still a problem with the fix, right? Would be cool if you could add a code comment which describes briefly what you changed to make it work :)

**Comment by philippotto** [comment]
I just tested the branch on https://improvetimetracking.webknossos.xyz/reports/timetracking but for me no tooltip is rendered? Does it work for you on the dev instance?

**Comment by MichaelBuessemeyer** [comment]
> I just tested the branch on https://improvetimetracking.webknossos.xyz/reports/timetracking but for me, no tooltip is rendered? Does it work for you on the dev instance?

No, it does not work for me too. The bug seems to appear only if you have few data rows :thinking:

File: frontend/javascripts/admin/time/time_line_chart_view.js (hunk 1 above)
Comment (MichaelBuessemeyer) [reply frontend/javascripts/admin/time/time_line_chart_view.js]: I added some more comments on that. I hope they are precise enough and not too long

**Comment by MichaelBuessemeyer** [comment]
> I just tested the branch on https://improvetimetracking.webknossos.xyz/reports/timetracking but for me, no tooltip is rendered? Does it work for you on the dev instance?

I found a workaround for this -> see my comments in the code

File: frontend/javascripts/admin/time/time_line_chart_view.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            format: timeAxisFormat,
+            minValue: dateRange[0].toDate(),
+            maxValue: dateRange[1].toDate(),
+          },
+          allowHtml: true,
+        }}
+        graph_id="TimeLineGraph"
+        chartPackages={["timeline"]}
+        width="100%"
+        height="600px"
```
Comment (MichaelBuessemeyer) [inline frontend/javascripts/admin/time/time_line_chart_view.js]: @philippotto: We need a fixed height as the workaround we found, that displays the chart at full height, leaves out the whole y-axis with the actual dates. Having a fixed height solves this. 
I also shortly tried to force the y-axis to be rendered below the chart but failed :/

**Comment by MichaelBuessemeyer** [comment]
@philippotto This PR should be ready for your next review. I hopefully fixed the bug that causes the tooltip, not to render,

**Comment by MichaelBuessemeyer** [comment]
@philippotto I think I still need your review on this PR.

**Pull request merged**